alire_common.gpr: remove default host os to avoid mistakes on macOS

Fabien-Chouteau opened the pull request:
With the default being Linux, macOS users will not notice any issue until they try to use the executable. Even then the issue will be obscure.

Fabien-Chouteau committed:
alire_common.gpr: remove default host os to avoid mistakes on macOS

With the default being Linux, macOS users will not notice any issue until they try to use the executable.
Even then the issue will be obscure.

Fabien-Chouteau commented:
This might trigger some build issues...

mosteo commented:
Then I think it will be better to remove reliance on `OS` and having a default as well from `alr_env.gpr`. Better to make it explicit everywhere?

Note to self: this will require fixing the `setup_alire` action for the case of building a branch.

mosteo committed:
Remove any implicit detection of the host OS

This should avoid misbuilds that mix Linux/macOS.

A bash function is now provided for detection of the OS in our scripts.

mosteo commented:
I've removed any guess attempts from `alr_env.gpr` too, and added a couple of bash functions for our use in tests/builds. Also reviewed a bit the README. Removed a couple of obsolete scripts and tweaked the rest.

I think we're good to go if you don't see any issue in my changes, @Fabien-Chouteau

Fabien-Chouteau commented:
Looks good to me :+1: Thanks for picking that up
